packages/worker/src/worker-options.ts

```diff
@@ -420,6 +426,47 @@ export function addDefaultWorkerOptions(options: WorkerOptions): WorkerOptionsWi
   };
 }
 
+/**
+ * Inspect the execution environment and return information about available system resources,
+ * taking into account applicable constraints that can be discovered.
+ * At present, it accounts for CGroups v1 and v2 constraints on memory and swap.
+ */
+export function inspectSystemResources(): { memory: number; heap: number } {
```

Comment:
At this point I'd consider just documenting this instead of issuing a warning but I don't have a strong preference either way.

Reply:
I agree that we can disregard the case where `--max-old-space-size` has been explicitly set to a somewhat high but _valid_ value.

However, I think the warning is desirable if the program is running under memory constraints and `--max-old-space-size` has not been set at all. This is _never_ a good situation. OOM is inevitable if the process lives long enough. Yet, most people out there using node and docker won't know about this fact.

So here is what I propose... I remove the percentage and rewrite the message to address specifically the situation I described above. Also, I refer user to the docs for more advice. Does that seems better?

```
    const hasMemoryConstraints = cgroupv1MemoryMax < resources.memory || cgroupv2MemoryMax < resources.memory;
    if (hasMemoryConstraints && resources.heap > resources.memory) {
      const memInMb = toMB(resources.memory, 0);
      const suggestedOldSpaceSizeInMb = toMB(resources.memory * 0.75, 0);

      Runtime.instance().logger.warn(
        `This program is running inside a containerized environment with a memory constraint ` +
          `(eg. docker --memory ${memInMb}m). node itself does not consider such memory constraint ` +
          `in how it manage its heap memory. There is consequently a high probability that ` +
          `the process might face OOM errors. To increase reliability, we recommend adding ` +
          `'--max-old-space-size=${suggestedOldSpaceSizeInMb}' to your node's arguments. ` +
          `Refer to https://docs.temporal.io/application-development/worker-performance for more ` +
          `advices on how to properly tune your workers.`
      );
    }
  }
```

Reply:
This is totally fine but TBH I'd be fine with documenting this too.
Your call, you already did most of the work so I guess it wouldn't hurt.